backend-rust/src/indexer.rs

```diff
@@ -4380,6 +4388,9 @@ impl PreparedPaydaySpecialTransacionOutcomes {
         &self,
         tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
     ) -> anyhow::Result<()> {
+        if !self.has_reward_events {
```

Comment:
Most of these are empty. Not really a reason why we would want to perform database operations on empty list